pkg/conversion/test/fuzz/fuzz.go

```diff
@@ -263,10 +270,28 @@ func MAPI2CAPIMachineRoundTripFuzzTest(scheme *runtime.Scheme, infra *configv1.I
 		// Do not match on status yet, we do not support status conversion.
 		// Expect(mapiMachine.Status).To(Equal(in.machine.Status))
 
-		Expect(mapiMachine.TypeMeta).To(Equal(in.machine.TypeMeta))
-		Expect(mapiMachine.ObjectMeta).To(Equal(in.machine.ObjectMeta))
-		Expect(mapiMachine.Spec).To(WithTransform(ignoreMachineProviderSpec, testutils.MatchViaJSON(ignoreMachineProviderSpec(in.machine.Spec))))
-		Expect(mapiMachine.Spec.ProviderSpec.Value.Raw).To(MatchJSON(in.machine.Spec.ProviderSpec.Value.Raw))
+		// Annotations propagation happens differently in MAPI vs CAPI, so the roundtrip is lossy.
+		// We document this in the migration docs.
+		Expect(mapiMachine.ObjectMeta.Annotations).To(equalMap(util.MergeMaps(in.machine.ObjectMeta.Annotations, in.machine.Spec.ObjectMeta.Annotations)), "converted MAPI machine should have matching .metadata.annotations")
+		Expect(mapiMachine.Spec.ObjectMeta.Annotations).To(equalMap(util.MergeMaps(in.machine.ObjectMeta.Annotations, in.machine.Spec.ObjectMeta.Annotations)), "converted MAPI machine should have matching .spec.annotations")
+		mapiMachine.ObjectMeta.Annotations = nil
+		mapiMachine.Spec.ObjectMeta.Annotations = nil
+		in.machine.ObjectMeta.Annotations = nil
+		in.machine.Spec.ObjectMeta.Annotations = nil
```

Comment:
Where we have done this before, we have fixed the issue at the fuzzer level. So where the two maps need to be the same on both sides, we could handle this in the custom fuzzer, and we don't need special logic changes here. That may be better as it'll be consistent with how we are handling other cases (in the fuzzer vs in the test checks)

Reply:
I don't think fuzzing is the right place to be testing something so nuanced, we should have specific tests.

The purpose of the fuzz tests predominantly is to pick up when new fields are added as we vendor changes, or where we have gaps in conversion as we introduce new platforms.

I think it's totally acceptable to say "we must have the same labels in two places in MAPI as the source of a fuzz" so that the fuzzing can focus on it's role, and have separate tests that handle the complexity of the labels conversion elsewhere

Do we have those separate tests right now?

Reply:
If we are going to add separate testing for labels then yeah, I think what you are saying makes sense.
If not, then we probably shouldn't.

But yes, having specific testing for this is probably going to be wiser.

Reply:
If you agree, I can update the fuzzer like you said in this PR, and then parallelise the work on conversion wide label testing in a follow up PR. WDYT?

Reply:
Yep, lets update the fuzzer and follow up with specific label testing in units please

Please make a card to track